ConfigSpace/configuration_space.pyx

```diff
@@ -1269,6 +1269,7 @@ class ConfigurationSpace(collections.abc.Mapping):
 
         unconditional_hyperparameters = self.get_all_unconditional_hyperparameters()
         hyperparameters_with_children = list()
+        conditional_hyperparameters = sorted(self.get_all_conditional_hyperparameters(), key=lambda hp_name: self._hyperparameter_idx[hp_name])
```

Comment:
n ... number of conditional hyperparameters
N ... number of all hyperparameters

This implementation has the complexity O(n log n d(N)), where d(N) is the complexity of fetching an element from a dictionary indexed by N strings.

Alternative:
`[hp_name for hp_name in self._hyperparameters if hp_name in self.get_all_conditional_hyperparameters()]`
Complexity: O(N s(n)), where s(n) is the complexity of checking membership in a set of n strings.